Revise Shutdown and Draining section (#1512)
getsentry/sentry-docs: src/collections/_documentation/error-reporting/configuration/drain-example/native.md

Calling `sentry_shutdown()` before exiting the application is critical to avoid
data loss.

> After shutdown the client cannot be used any more so make sure to only do that

Reviewer: ```suggestion
After shutdown, the client cannot be used anymore, so make sure to do that
```

Author: Same concern re: "only".

"make sure to do something _only_ [at this given time]"

vs

"make sure to do something [at this given time]"

Reviewer: How about:

After shutdown, the client cannot be used anymore. It's important to only call `sentry_shutdown()` immediately before shutting down the application.

Author: Suggestion applied, thank you!

> right before you shut down the application.

Author: This is moved in here from the global context with no changes.

As for the general contents of this file, @jan-auer @Swatinem is there anything else you'd like to add?
Is this still accurate and reflecting the best practice?

> right before you shut down the application.

Reviewer: ```suggestion
 before you shut down the application.
```

Author: Thinking about "right before" vs "before".

In this case, "before" sounds like any time before -- not so precise.
Conversely, I think the intention of "right before" was to mean "call `close` and the very next thing is the end of the app". The latter is the correct, precise suggestion.

@MimiDumpling could you elaborate on why we would remove "right"?

Reviewer: Made a suggestion in the comment above. 😄 

The word "right" isn't quite precise enough in this instance. "Right" has many meanings. I think a better word for this case is "immediately."
